quickwit/quickwit-indexing/src/actors/indexing_service.rs

```diff
@@ -127,7 +112,7 @@ pub struct IndexingService {
     indexing_directories: HashMap<(IndexId, SourceId), WeakIndexingDirectory>,
     local_split_store: Arc<LocalSplitStore>,
     max_concurrent_split_uploads: usize,
-    merge_pipeline_handles: HashMap<MergePipelineId, MergePipelineHandle>,
+    merge_pipeline_handles: HashMap<(IndexId, SourceId), MergePipelineHandle>,
```

Comment:
Since we are going to keep `MergePipelineId`, the previous `HashMap<MergePipelineId, MergePipelineHandle>` was better.